streamer/pulp/streamer/server.py

```diff
@@ -378,3 +376,60 @@ def write(self, data):
         :type  data: str
         """
         reactor.callFromThread(self.request.write, data)
+
+
+class SessionCache(Cache):
+    """
+    Session cache.
+    Extends generic cache to derive the key using the URL and
+    downloader configuration.
+    """
+
+    @staticmethod
+    def key(url, downloader):
+        """
+        Get a cache key for the downloader.
+
+        :param url: The download URL.
+        :type url: basestring
+        :param downloader: A file downloader.
+        :type downloader: nectar.downloaders.base.Downloader
+        :return: A cache key based on downloader configuration.
+        """
+        url = urlparse(url)
+        properties = (
+            url.hostname,
+            url.port,
+            downloader.config.basic_auth_username,
+            downloader.config.basic_auth_password,
+            downloader.config.proxy_url,
+            downloader.config.proxy_port,
+            downloader.config.proxy_username,
+            downloader.config.proxy_password,
+            downloader.config.ssl_client_cert
+        )
+        return hash(properties)
+
+    def get(self, url, downloader):
+        """
+        Get a cached session for the url and downloader.
+        A session is created and configured when not already cached.
+
+        :param url: The download URL.
+        :type url: str
+        :param downloader: A nectar downloader.
+        :type downloader: nectar.downloaders.core.Downloader
+        :return: A cached session.
+        :rtype: Session
+        """
+        key = self.key(url, downloader)
+        try:
+            session = super(SessionCache, self).get(key)
+        except NotCached:
+            adapter = HTTPAdapter()
+            session = Session()
+            session.stream = True
+            session.mount('http://', adapter)
```

Comment:
Is this required if we're using the default adapter? I thought this would only be required if we were providing a customized adapter.